src/Handler/Bucket/Create/GrantBucketAccessToReadOnlyRoleHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\BigQuery\Handler\Bucket\Create;
+
+use Google\ApiCore\ApiException;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\DestinationDataset;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\DestinationDatasetReference;
+use Google\Protobuf\Internal\Message;
+use Keboola\StorageDriver\BigQuery\CredentialsHelper;
+use Keboola\StorageDriver\BigQuery\GCPClientManager;
+use Keboola\StorageDriver\BigQuery\NameGenerator;
+use Keboola\StorageDriver\Command\Bucket\GrantBucketAccessToReadOnlyRoleCommand;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Shared\Driver\Exception\Exception;
+
+final class GrantBucketAccessToReadOnlyRoleHandler implements DriverCommandHandlerInterface
+{
+    public GCPClientManager $clientManager;
+
+    public function __construct(GCPClientManager $clientManager)
+    {
+        $this->clientManager = $clientManager;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof GrantBucketAccessToReadOnlyRoleCommand);
+
+        assert(
+            $command->getProjectReadOnlyRoleName() !== '',
+            'GrantBucketAccessToReadOnlyRoleCommand.projectReadOnlyRoleName is required'
+        );
+        assert(
+            $command->getBucketObjectName() !== '',
```

Comment:
$command->getProjectReadOnlyRoleName(), $command->getBucketObjectName() mozno este prehodim v kontextu bq to obsahuje ine veci ako sa teraz aktualne poosielaju z connection.

getProjectReadOnlyRoleName - sa pouziva pre  meno datasetu (bucketu)
getBucketObjectName - sa pouziva pre listing vytvoreneho pre externy bucket, a posle sa v parametre pri registacii.

Reply:
Jj tak som to myslel,

$command->getProjectReadOnlyRoleName() = listing
$command->getBucketObjectName() = meno linknuteho bucketu v cielovom projekte